pkg/kstatus/status/core.go

```diff
@@ -345,6 +355,38 @@ func daemonsetConditions(u *unstructured.Unstructured) (*Result, error) {
 	}, nil
 }
 
+// checkDaemonsetGeneration checks that the metadata.generation and
+// status.observedGeneration fields are set.
+func checkDaemonsetGeneration(u *unstructured.Unstructured) (*Result, error) {
```

Comment:
Could we find a more generic name here? The same issue is almost certainly true for many other resources as well, including deployments and statefulset.